discovery-provider/src/tasks/cache_user_balance.py

```diff
@@ -272,8 +272,20 @@ def refresh_user_ids(
                         )
                         waudio_balance = bal_info["result"]["value"]["amount"]
 
+
                 # update the balance on the user model
                 user_balance = user_balances[user_id]
+
+                # Write to user_balance_changes table
+                session.add(
+                    UserBalanceChange(
+                        user_id=user_id,
+                        blocknumber=Web3.eth.blockNumber,
+                        current_balance=user_balance.balance,
+                        previous_balance=owner_wallet_balance,
```

Comment:
I think we want to store the sum of the user's balance in their wallet, associated wallet, PDA token account and associated solana wallets - note that the PDA and associated solana wallets have a different number of decimals.

Reply:
ah I thought the balance was the sum. will update 👍